Subscribers/Lifecycle.php

```diff
@@ -167,6 +169,29 @@ public function onDeleteArticle(\Enlight_Event_EventArgs $eventArgs)
     }
 
     /**
+     * Callback function to delete an product from connect
+     * after it is going to be deleted locally
+     *
+     * @param \Enlight_Event_EventArgs $eventArgs
+     */
+    public function onPreDeleteCategory(\Enlight_Event_EventArgs $eventArgs)
+    {
+        $category = $eventArgs->get('entity');
+        $this->connectExport->markProductsInToBeDeletedCategories($category);
```

Comment:
here you can store category id in private property. ```$this->removedCategoryId = $category->getId()```. then methods are stateful, but cron_update is used for something difference. Then ```$this->connectExport->markProductsInToBeDeletedCategories($category);``` can be removed.